### PR TITLE
feat/779 - SDK: Incorporate Tx batching functionality

### LINKED ISSUE #779: Extension/Interface: Update to support Tx batching
Now that https://github.com/anoma/namada/pull/3103 is merged into `v0.36.1`, we can make use of batch Tx instead of building, signing, and issuing multiple Tx sequentially (this will save gas and should simplify SDK JS API)

(_NOTE_ Will add more details here as I research how to implement this!)

- Tx should be built as a batch always, and we can consider a single Tx to just be a batch of 1 item (in `packages/shared`)
- SDK JS implements this as a part of the API (in `packages/sdk`)
- Interface is updated to support this change, and the work-around in fee calculation removed as it will not be needed (we do not need to multiple fee for a tx by number of tx)

### PR DESCRIPTION
**Pull request by jurevans**
Resolves #779 

- [x] Introduce support for batch Tx in `shared` & `sdk`
- [x] Incorporate this step into interface Tx
- [x] Sign batch Tx correctly and confirm each Tx is applied to chain
- [x] Update `process_tx` response (and `rpc.broadcastTx` response in SDK) to allow clients to see status of individual Tx within the batch

### CODE REVIEW

File: apps/namadillo/src/lib/query.ts
Comment (emccorson) [inline apps/namadillo/src/lib/query.ts:101]: I think this will compile without casting with `as BuiltTx`, and also in the call to `buildBatch` below.

File: apps/namadillo/src/lib/query.ts
Comment (emccorson) [inline apps/namadillo/src/lib/query.ts:94]: Just a question, but is there a reason for not including the reveal PK TX in the batch?

**Review by emccorson — APPROVED** [review]
It works well and I just left a couple of small comments. I'm not super familiar with the TX code though so I would get a second review :smile:

File: apps/namadillo/src/lib/query.ts
Comment (jurevans) [reply apps/namadillo/src/lib/query.ts:94]: Yes! I separated reveal PK so that the batch contains one Tx type, so it has its own approval for now to make it clear that it is a Tx that needs to be approved and sent. Maybe we would want to include it, but for displaying details and such, it's much easier to think about if a batch is of one Tx type

File: apps/namadillo/src/lib/query.ts
Comment (jurevans) [reply apps/namadillo/src/lib/query.ts:101]: I will take a look! I'd really like to get rid of this `BuiltTx` entirely and clean up some of this Tx code. I feel it is unnecessarily complicated

File: apps/namadillo/src/lib/query.ts
Comment (jurevans) [reply apps/namadillo/src/lib/query.ts:94]: I think there is no technical reason that would prevent us from adding it into the batch, but from a UX point of view, the interface always sends a batch of a single tx type at a time. If we decide we want to do it, it will be trivial to add it in. I will ask!

File: apps/namadillo/src/lib/query.ts
Comment (jurevans) [reply apps/namadillo/src/lib/query.ts]: ah, yeah, I was messing with having a separate type for batches but removed it, so that was something I forgot to change back :)

File: apps/extension/src/background/keyring/messages.ts
Comment (pedrorezende) [inline apps/extension/src/background/keyring/messages.ts:355]: Sorry if I'm missing something, but are we using QueryAccountDetailsMsg anywhere? I've noticed this message type hasn't been added to the handler function

**Review by pedrorezende — APPROVED** [review]
LGTM

File: apps/extension/src/background/keyring/messages.ts
Comment (jurevans) [reply apps/extension/src/background/keyring/messages.ts:355]: You're right! This was copied from my Ledger update PR, I can remove it here

Comment (jurevans) [reply apps/extension/src/background/keyring/messages.ts]: Fixed!

**Pull request merged**
